Sources_v3/ChatClient_Tests.swift

```diff
@@ -483,6 +483,70 @@ class ChatClient_Tests: StressTestCase {
             Assert.willBeTrue(connectCompletionCalled)
         }
     }
+
+    // MARK: - Creating `ChannelController` tests
+
+    func test_channelControllerForNewChannel_createdCorreclty() {
```

Comment:
Please put these tests to `ChannelController_Tests`. Even though the functions are called in the client, they are related to `ChannelController`.

Reply:
Are you sure about this one?
I've thought about this and I believe it can become confusing in future if we will make decisions on test location based on some relations. 
By this logic some tests from `ChannelController` probably will be moved to `ChannelUpdater`.
It seems more straightforward if tests of `ChatClient` methods is in `ChatClient_Tests` even though they are in extension in `ChannelController` file.

Reply:
I see what you mean but this is a special case. You can think of these functions in the `ChatClient` extension as of convenience initializers. The only reason why they are not named `convenience init` is that we want to provide a better public API and enforce a certain level of coupling for the users. We want them to use the client instance to create all controllers.

So, while your comment is valid and makes a total sense for any other scenario, this is a special case.